### ISSUE TITLE
Label masks configuration not loading from style QML

### ISSUE THREAD

**Issue by dinizime** [qgis/QGIS#43710]
_Tested on Windows 10 / QGIS 3.18.3._

When saving a style to QML, "Masks" is not available to select.

![image](https://user-images.githubusercontent.com/10109704/121814500-ce405900-cc47-11eb-8fbb-312aab66cfb8.png)

However if you open the QML on a text editor there is a *maskedSymbolLayers* property in the label configuration.

If you load this QML it does not update the Mask configuration for the layer.

I did two custom scripts to save and load the masks configuration.

Save:
https://gist.github.com/dinizime/eb4ef5c7ac195b935fc3ac1a75812d55
Load:
https://gist.github.com/dinizime/2fae3925802e5ce59a94db94fb5af1a9

However, it looks like the ruleKey for QgsRuleBasedLabeling rules changes when the layer loads. The way to circumvent this is to reference the rules from QgsRuleBasedLabeling by description, and not by the ruleKeys.

**Comment by roya0045** [comment]
Unsure if this is a bug since masking depends on a lot of element within a project and reusing that could cause some issues IMO.

**Comment by dinizime** [comment]
> Unsure if this is a bug since masking depends on a lot of element within a project and reusing that could cause some issues IMO.

I agree that it can be too complicated for reusing this element, and my script is enough for my use case. However, it would be better if the ruleKey for the QgsRuleBasedLabeling was fixed.

**Comment by Pedro-Murteira** [comment]
Still valid on QGIS 3.22.3.
